irony-cdb.el

```diff
@@ -114,6 +128,24 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--locate-dominating-file-with-dirs (file name subdirectories)
+  "Convenience wrapper around `locate-dominating-file'
+
+Looks up the directory hierarchy from FILE for to locate any directory
+in `subdirectories` which contains NAME. Stops at the first parent
+directory where the test succeeds and returns the full path to file
+found."
+  (let ((check-subdirs
```

Comment:
Ok, I think I'm super-confused by the 2 appearances of `check-subdirs`.

IIUC you are now using the version of `locate-dominating-file` where name is a function and not a path. I would prefer to avoid using this one since it is only available in Emacs 24. The last Ubuntu LTS is still using Emacs 23.4, we should support it.